src/scales/scale.time.js

```diff
@@ -562,6 +562,37 @@ module.exports = function(Chart) {
 			me._minorFormat = formats[unit];
 			me._majorFormat = formats[majorUnit];
 			me._table = buildLookupTable(me._timestamps.data, min, max, options.distribution);
+			me._leftOffset = 0;
+			me._rightOffset = 0;
+
+			if (options.offset && ticks.length) {
```

Comment:
I haven't tested, but we cannot simply use `me._timestamps.data` to compute offsets because the array can contain timestamps from different types of datasets for example line with the point type and bars.